R/taxonomy.R

```diff
@@ -209,6 +219,25 @@ setMethod("checkTaxonomy", signature = c(x = "SummarizedExperiment"),
     }
 )
 
+#' @rdname taxonomy-methods
+#'@export
+# Function to set taxonomy ranks
+setTaxonomyRanks <- function(rank) {
+    rank <- tolower(rank)
+    # Check if rank is a character vector with length >= 1
+    if (!is.character(rank) || length(rank) < 1  || any(rank == "" | rank == " " | rank == "\t" | rank == "-" | rank == "_") || any(grepl("\\s{2,}", rank))) {
+        stop("Input 'rank' should be a character vector with non-empty strings, no spaces, tabs, hyphens, underscores, and non-continuous spaces.")
```

Comment:
Add here call. = FALSE

Reply:
Run also BiocCheck --> indentation should be multiplication of 4 spaces and the width of line should be less than 80 characters.